examples/ippserver.rs

```diff
@@ -2,81 +2,322 @@ extern crate hyper;
+                           IppAttribute::new(JOB_STATE_REASONS,
+                               IppValue::Keyword("completed-successfully".to_string())));
+
+        self.printing.store(true, atomic::Ordering::Relaxed);
+        let mut file = OpenOptions::new().write(true).create(true).open("printjob.dat").unwrap();
+        io::copy(&mut req.req, &mut file).unwrap();
+        Ok(resp)
     }
 
-    fn create_job<'a>(&self, _req: &mut IppRequestResponse) -> IppServerResult<'a> {
+    fn validate_job<'a>(&self, req: &mut Self::IppRequest) -> IppServerResult<'a> {
+        println!("Validate-Job");
+        println!("{:?}", req.header());
+        println!("{:?}", req.attributes);
+        println!("");
+        let resp = IppRequestResponse::new_response(StatusCode::SuccessfulOK as u16,
+                                                        req.header().request_id);
+
+        Ok(resp)
+    }
+
+    fn create_job<'a>(&self, _req: &mut Self::IppRequest) -> IppServerResult<'a> {
         Err(StatusCode::ServerErrorOperationNotSupported)
     }
 
-    fn cancel_job<'a>(&self, _req: &mut IppRequestResponse) -> IppServerResult<'a> {
+    fn cancel_job<'a>(&self, _req: &mut Self::IppRequest) -> IppServerResult<'a> {
         Err(StatusCode::ServerErrorOperationNotSupported)
     }
 
-    fn get_job_attributes<'a>(&self, _req: &mut IppRequestResponse) -> IppServerResult<'a> {
+    fn get_job_attributes<'a>(&self, _req: &mut Self::IppRequest) -> IppServerResult<'a> {
         Err(StatusCode::ServerErrorOperationNotSupported)
     }
 
-    fn get_jobs<'a>(&self, _req: &mut IppRequestResponse) -> IppServerResult<'a> {
+    fn get_jobs<'a>(&self, _req: &mut Self::IppRequest) -> IppServerResult<'a> {
         Err(StatusCode::ServerErrorOperationNotSupported)
     }
 
-    fn get_printer_attributes<'a>(&self, req: &mut IppRequestResponse) -> IppServerResult<'a> {
+    fn get_printer_attributes<'a>(&self, req: &mut Self::IppRequest) -> IppServerResult<'a> {
+        const SUPPORTED_ATTRIBUTES : [&'static str; 23] = [
+            PRINTER_URI_SUPPORTED,
+            URI_SECURITY_SUPPORTED,
+            URI_AUTHENTICATION_SUPPORTED,
+            PRINTER_NAME,
+            PRINTER_STATE,
+            PRINTER_STATE_REASONS,
+            IPP_VERSIONS_SUPPORTED,
+            OPERATIONS_SUPPORTED,
+            CHARSET_CONFIGURED,
+            CHARSET_SUPPORTED,
+            NATURAL_LANGUAGE_CONFIGURED,
+            GENERATED_NATURAL_LANGUAGE_SUPPORTED,
+            DOCUMENT_FORMAT_DEFAULT,
+            DOCUMENT_FORMAT_SUPPORTED,
+            PRINTER_IS_ACCEPTING_JOBS,
+            QUEUED_JOB_COUNT,
+            PDL_OVERRIDE_SUPPORTED,
+            PRINTER_UP_TIME,
+            COMPRESSION_SUPPORTED,
+            PRINTER_STATE_MESSAGE,
+            PRINTER_MAKE_AND_MODEL,
+            FINISHINGS_DEFAULT,
+            FINISHINGS_SUPPORTED,
+        ];
+        let supported_attributes = &SUPPORTED_ATTRIBUTES[..];
+
```

Comment:
I think lazy_static macro (https://github.com/rust-lang-nursery/lazy-static.rs) and vec! would be more suitable here.